spec/depth-enabled-photography.html

```diff
@@ -2160,121 +2160,417 @@
+              function(d) {
+                console.log(
+                    'Distance between(' + startX + ',' + startY +
+                    ') - (' + endX + ',' + endY + ') = ' +
+                    parseFloat(d.distance).toFixed(2) + ' millimeters, Confidence = ' +
+                    parseFloat(d.confidence).toFixed(2) + ', Precision=' +
+                    parseFloat(d.precision).toFixed(2) + 'mm');
+              },
+              function(error) { console.log(error.message); });
+
+
+          // Motion Effect
+          var motionEffect;
+          try {
+            motionEffect = new realsense.DepthEnabledPhotography.MotionEffect();
+          } catch (error) {
+            console.log(error.message);
+            return;
+          }
+          motionEffect.init(photo).then(
+              function() {
+                doMothionEffect();
+              },
+              function(error) { console.log(error.message); });
+
+          function doMothionEffect() {
+            var right = 0.0, up = 0.0, forward = 0.0;
+            var yaw = 0.0, pitch = 0.0, roll = 0.0;
+            var zoom = 0.0;  // Set these motion parameters at your convenience.
+            motionEffect.apply({ horizontal: right, vertical: up, distance: forward },
+                               { pitch: pitch, yaw: yaw, roll: roll },
+                               zoom).then(
+                function(image) {
+                  // Render the motion effect applied photo image.
+                },
+                function(error) { console.log(error.message); });
+          }
+
+
+          // Segmentation
+          var segmentation;
+          try {
+            segmentation = new realsense.DepthEnabledPhotography.Segmentation();
+          } catch (error) {
+            console.log(error.message);
+            return;
+          }
+
+          var markupImage = {
+            format: 'Y8',
+            width: photo.width,
+            height: photo.height,
+            data: markupImageBuffer // An ArrayBuffer for bounding mask image data
+                                    // (image.width x image.height), pixel value should be
+                                    // 255 for foreground and 0 for background.
+          };
+          segmentation.objectSegment(photo, markupImage).then(
+              function(maskImage) {
+                photo.queryContainerImage().then(
+                    function(colorImage) {
+                      // Blend and render the colorImage and maskImage;
+                    },
+                    function(error) { console.log(error.message); });
+              },
+              function(error) { console.log(error.message); });
+
+          var maskArea = [{x: 0, y: 0}, {x: 10, y: 10}, ...]; // Define the bounding mask area.
+          var isForeground = true; // Indicate if the refinement take place in forground or background.
+          segmentation.refineMask(maskArea, isForeground).then(
```

Comment:
This function should be called after `segmentation.objectSegment` succeeds with an initial mask image. You may use an event handler and comments to give audience the idea that the `maskArea` is usually created with user interaction on blending result of `colorImage` and previous mask.